docs/build/wasm/transaction-fees.md

```diff
@@ -0,0 +1,147 @@
+---
+sidebar_position: 2
+---
+
+# Transaction Fees
+
+## Weight
+
+As is also the case with Substrate, `pallet-contracts` uses [weightV2][weight] to charge execution fees. It is composed of `refTime` and `proofSize` :
+- refTime: The amount of computational time that can be used for execution, in picoseconds.
+- proofSize: The amount of storage space that can be used (also called storage bandwidth), in bytes. So access storage assume that it will grow the gas fees.
+
+:::info
+Gas = Weight = (refTime, proofSize)
+:::
+
+[Transaction Weight in Substrate Documentation][weight]
+
+## Storage Rent
+
+Storage rent, also called as *Automatic Deposit Collection* is a **mechanism** that ensure security of the chain by preventing on-chain storage spamming.
+It prevents malicious actors from spamming the network with low-value transactions and to ensure that callers have a financial stake when storing data on-chain.
+
+Users will be charged for every byte stored on-chain and the call will transfer this fee from the free balance of the user to the reserved balance of the contract. Note that the contract itself is unable to spend this reserved balance (but it can expose a function that remove on-chain storage and the caller will get the funds) .
+It also incentives users to remove unused data from the chain by getting rent fees back. Any user can get back the rent fees if they remove on-chain data (not specifically the user that was first charged for).
```

Comment:
I'd add something like - "It's up to the contract developer & users to understand how and if they can get their storage deposit back."